hw/ip/otbn/dv/otbnsim/sim/stats.py

```diff
@@ -35,6 +42,11 @@ def record_stall(self) -> None:
         '''Record a single stall cycle.'''
         self.stall_count += 1
 
+    def _insn_at_addr(self, addr: int) -> Optional[OTBNInsn]:
+        '''Get the instruction at a given address.'''
+        word_addr = addr >> 2
```

Comment:
Does this want an assertion that `addr` is non-negative and divisible by 4 and that `word_addr` is in range?

Reply:
Asserts are added, the out of range check will be implicitly done by the array access. (Since the address has already been checked multiple times before, throwing an exception here seems fine to me. It's Python, after all :-))